### DISCUSSION TITLE
Connect To A Supysonic Server

### DISCUSSION THREAD

**Discussion by PeterRob** [jeffvli/sonixd#63] [Q&A]
I'm trying to connect to a supysonic server proxied by caddy by HTTPS. Error message is 'A Required Parameter Is Missing'.

Sonixd seems to expect a specified port no in the URL field (I tried 443 with same error message).

[https://photos.app.goo.gl/yoM9fGmYwLhzEyzR8](url)

**Comment by jeffvli** [comment]
Hmm, I haven't tested with supysonic, but from the error message it seems like it's actually the server that is rejecting the login due to a missing parameter on the actual request.

Let me spin up a server to test and I'll see if I can fix that.

**Comment by jeffvli** [comment]
Seems like supysonic doesn't support the hashed authentication method. I'll add a legacy option (plaintext) on the login screen for the next release ~~(probably in 1-2 days)~~. 

Edit: This will take a bit longer than expected. I've already added the functionality in the dev branch if you want to build it yourself, but I want to add a couple more features before pushing 0.5.0.

**Reply by jeffvli** [reply]
@PeterRob Legacy auth has been added in [0.5.0](https://github.com/jeffvli/sonixd/releases/tag/v0.5.0). I actually don't have a working supysonic server to test on, so let me know if you find any broken features.

**Reply by PeterRob** [reply]
I tested 0.5.0 connecting with supysonic-web running from the command line, using legacy auth. The password seems to be truncated:

`[pid: 23486|app: 0|req: 25536/25536] 127.0.0.1 () {38 vars in 765 bytes} [Sat Nov  6 16:49:42 2021] GET /rest/getScanStatus?v=1.15.0&c=sonixd&f=json&u=peter&p=SjTU => generated 121 bytes in 6 msecs (HTTP/1.1 200) 3 headers in 104 bytes (1 switches on core 0)`

Error message 'Wrong username or password'

**Reply by jeffvli** [reply]
Strange.. I don't see the password being truncated when testing on my side.

If at all possible would you be able to run the application in a dev environment following the [startup instructions](https://github.com/jeffvli/sonixd#development--contributing)? That would allow you to see the network requests being made in the browser devtools (f12).

**Reply by PeterRob** [reply]
I'll give it a try and revert. Cheers!

**Reply by PeterRob** [reply]
I tested in the dev environment, successfully logged in with a pass phrase (23 chars). 

My original password (SjTU#9tGL@58!cTPE&WmW3P4PQabEGmU) included a '#', as you can see from the supysonic-web output, the '#' is not passed by sonixd.

Now I can log in, will try the app - thanks!

**Reply by jeffvli** [reply]
Glad you figured it out and enjoy! I'll check on my side to see why the `#` character is being truncated.
